fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1617,7 +1629,7 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
                 # unsharded gradients allocated; one for a pending reduction, and one for gradient computation.
                 param.grad = None
                 callback_fn = functools.partial(self._post_reduction_hook, param)
-                grad_chunks = chunk_and_pad(grad, self.world_size)
+                grad_chunks = chunk_and_pad(grad, self.process_group_reduce_scatter.size())
```

Comment:
Can you add a test for this?

Reply:
All the unit tests are still passing. Our unittest env only have 2 gpus. It is difficult to create default process group and reduce scatter process group with different number of ranks. By default, both default process group and reduce scatter process group are using all the ranks in dist. The size of ranks are the same.  #904 probably uses unconventional way to initiate the default process group.